app/controllers/posts_controller.rb

```diff
@@ -438,7 +438,7 @@ def restore
       return
     end
 
-    if @post.deleted_by.is_moderator && !current_user.is_moderator
+    if @post.deleted_by.at_least_moderator? && !current_user&.at_least_moderator?
```

Comment:
Untested suggestion:

```suggestion
    if @post.deleted_by.at_least_moderator? && current_user&.standard?
```

Reply:
Same thing here - resolving as we decided to avoid `!current_user || current_user.standard?`, which is needed due to the safe navigation operator.